src/main/java/wellnus/focus/feature/Session.java

```diff
@@ -26,19 +40,21 @@ public class Session {
      */
     public Session() {
         this.session = new ArrayList<>();
+        initialiseSession();
+        this.currentCountdownIndex = session.size() - INCREMENT;
```

Comment:
Can I clarify this line will set currentCountdownIndex to the last index?

Reply:
Yes, this will set the current countdown index to the last index.

I'll explain the thought process here for documentation sake and for you to check if it makes sense!

This behaviour due to your initial implementation of the code where increments will only occur on `start` and `next`.

To have this logic work without excessive use of flags and passing messages between components, I thought that the common behaviour for `start`ing a session the 1st time and nth time would naturally be to have the pointer at the last countdown with the `isReadyFlag`. 

The idea is that when a session ends, the `currentCountdownIndex` pointer will always point to the last countdown object. Since increments are only done in `start` and `next` commands as per your implementation, I need a way to differentiate between the two using the `isReady` flag. 

To make this behaviour invariant,the  `stop` command also forces `currentCountdownIndex` to point to the last countdown object and toggle its `isReady` flag to true.

```java
// starting from StopCommand line 81
session.getCurrentCountdown().setStop();
sessioninitialiseSession();
session.resetCurrentCountdownIndex();
```

I have experimented with making the timer 1-indexed and even having the flag on , but without massive workarounds to have flags being passed between `Countdown` and `Session`, it is unfeasible. I felt that it would be much better to work with the code you have implemented already since it works! 👍🏼